### PR TITLE
Added table of content to open layer details in map view.

### LINKED ISSUE #1616: Showing metadata per layer in map
**Problem:** When I, as a viewer, look at a map with multiple data layers, I do not get detailed information about the layers used in the map without leaving it. So far, I can look at the map, but this jumps to the respective data set and the map is exited.

grafik

**Objective:** As a viewer, I would like to be able to display the detail panel of the layer on the right-hand side via an additional layer on the right side. For this, I do not leave the map view. A mockup is shown in the following figure.

grafik

### LINKED ISSUE #1660: No option to view detail of a layer from the map view
Problem: When I, as a viewer, look at a map with multiple data layers, I do not get detailed information about the layers used in the map without leaving it. So far I can look at the map, but this jumps to the respective data set and the map is left. is exited.

267068618-074e18fc-4ebb-41c1-a8fe-6df75c0fecb4

Objective: As a viewer, I would like to be able to display the detail panel of the layer on the right-hand side via an additional layer on the right side. For this I do not leave the map view. A mockup is shown in the following figure.

267068772-3456e3c9-858d-426f-9db6-4f76a0648b97

### PR DESCRIPTION
**Pull request by ahmdthr**
Resolves #1660
Resolves #1616

A new button to view layer details is added on the map view.

### CODE REVIEW

**Comment by ridoo** [comment]
@ahmdthr the PR still is marked as draft .. if this is intended, though, please ignore my comment

**Comment by ridoo** [comment]
@ahmdthr this one would have to be aligned to the latest changes of linked resources

**Comment by ridoo** [comment]
Thanks @ahmdthr 

@giohappy the PR is now ready for review on your side

**Comment by dsuren1** [comment]
@ahmdthr Kindly resolve the conflicts and align the code changes with the latest changes in master. Thanks!

**Comment by gannebamm** [comment]
@giohappy Sorry @ahmdthr was busy with other tasks and we need some additional days to sort out our ressources. Are we to late for lending a helping hand? @dsuren1 already did some changes and requested a review.

**Comment by giohappy** [comment]
I've tested the PR locally. The idea is good and it's really useful, but I have a few concerns:

 - every time a layer is selected in the TOC a new API request is sent to obtain the required information to enable (or not) the info button for the layer
 - the information that has been retrieved is not saved, so the same request is sent again every time a layer is selected

I think this could be improved if the following is implemented:

 - The response can be stored, somehow, inside the state of the component. This avoids making the same request for the same layer multiple times
 - But, the real point is: **are these calls really needed**? We already query both the map and the datasets endpoint when the map is loaded. Much information about datasets is retrieved at that time. Isn't this enough? If more attributes are required we can add them to the response of those requests, instead of making additional calls.

@gannebamm @ahmdthr if you agree I would consider these improvements before merging it. In that case the PR should be marked as `Draft` for the time being.

**Review by giohappy — CHANGES_REQUESTED** [review]
see https://github.com/GeoNode/geonode-mapstore-client/pull/1661#issuecomment-2117753350

**Comment by ahmdthr** [comment]
> I've tested the PR locally. The idea is good and it's really useful, but I have a few concerns:
> 
> * every time a layer is selected in the TOC a new API request is sent to obtain the required information to enable (or not) the info button for the layer
> * the information that has been retrieved is not saved, so the same request is sent again every time a layer is selected
> 
> I think this could be improved if the following is implemented:
> 
> * The response can be stored, somehow, inside the state of the component. This avoids making the same request for the same layer multiple times
> * But, the real point is: **are these calls really needed**? We already query both the map and the datasets endpoint when the map is loaded. Much information about datasets is retrieved at that time. Isn't this enough? If more attributes are required we can add them to the response of those requests, instead of making additional calls.
> 
> @gannebamm @ahmdthr if you agree I would consider these improvements before merging it. In that case the PR should be marked as `Draft` for the time being.

@giohappy Thank you very much for your comments. The only reason a new request was being made was because the linked resources were not included in the batch request. Now with recent changes, it is no longer necessary to sent a request to fetch linked resources. I am currently working on it and expect the final version of the PR shortly.

**Comment by ahmdthr** [comment]
The info button for layers on the maps page now fetches all data including linked resources with a single call. Additionally, the linked resources are stored in state such that re-fetching is not necessary when re selecting one layer again.

**Comment by giohappy** [comment]
@dsuren1 please do a new review.

File: geonode_mapstore_client/client/js/epics/index.js [hunk 1]
```diff
@@ -55,6 +56,33 @@ export const gnCheckSelectedDatasetPermissions = (action$, { getState } = {}) =>
...
+ * Fetches missing values for selected layers
+ */
+export const gnFetchMissingLayerData = (action$, { getState } = {}) =>
+    action$.ofType(SELECT_NODE)
+        .filter(({ nodeType }) => nodeType && nodeType === "layer")
+        .switchMap(() => {
+            const state = getState() || {};
+            const layer = getSelectedLayer(state);
+            const layerResourceId = layer?.extendedParams?.pk;
+            const layerResourceDataset = state.gnresource.data.maplayers.find(mapLayer => mapLayer.dataset.pk === parseInt(layerResourceId, 10)).dataset;
```
Comment (dsuren1) [inline geonode_mapstore_client/client/js/epics/index.js:69]: Error when `data` is empty

**Review by dsuren1 — CHANGES_REQUESTED** [review]
@ahmdthr 
It seems in some cases, an error is being generated in `gnFetchMissingLayerData` when selecting a layer, which is impacting further testing and review of this PR. 

image

Could you please check and update the PR? Thank you!

**Comment by ahmdthr** [comment]
@dsuren1 Thank you very much for your review. All the problems stemmed from non-existent referencing of properties, and I have fixed it in the latest commit. Please have a look at it again and let me know. Thank you.

File: geonode_mapstore_client/static/mapstore/configs/localConfig.json [hunk 2]
```diff
@@ -2330,6 +2330,150 @@
                 "cfg": {
                     "wrap": true
                 }
+            },
+            {
+                "name": "LayerDetailViewer",
```
Comment (dsuren1) [inline geonode_mapstore_client/static/mapstore/configs/localConfig.json:2335]:
```suggestion
                "mandatory": true,
                "name": "LayerDetailViewer",
```

In order to make the plugin available by default add `mandatory: true`, or add it to pluginsConfig under `TOC` as `children`, so the user can select the plugin when adding `viewer` to map.

Currently, the plugin is not available on map with viewer existing already.

**Review by dsuren1 — CHANGES_REQUESTED** [review]
@ahmdthr 
Overall the changes looks good. But I see some minor issues as mentioned below. Kindly check it out

- Layer detail doesn't always reflect correct layer name on the detail panel

  https://github.com/user-attachments/assets/3fbcb9b8-3d02-4e18-bb4c-bbafd80f2b82

**Comment by ahmdthr** [comment]
> @ahmdthr Overall the changes looks good. But I see some minor issues as mentioned below. Kindly check it out
> 
> * Layer detail doesn't always reflect correct layer name on the detail panel
>   
>     
>       
>       
>   
>       layer_detail_sync.mp4

This bug where the title of the dataset on the detail layer was not being changed was due to the fact that the "EditTtile" component was using its own state for the title, thus changing the "resource" did not update the title if the "DetailsPanel" was already inflated. It is fixed now.

**Review by dsuren1 — APPROVED** [review]

**Comment by ridoo** [comment]
@dsuren1 great to see you approved the PR .. any chance to see this PR backported to `4.4.x`?

**Pull request merged**